executor/src/witgen/machines/block_machine.rs

```diff
@@ -126,14 +128,15 @@ impl<'a, T: FieldElement> BlockMachine<'a, T> {
         connecting_identities: &BTreeMap<u64, &'a Identity<Expression<T>>>,
         identities: &[&'a Identity<Expression<T>>],
         witness_cols: &HashSet<PolyID>,
+        degree: u64,
```

Comment:
You don't even need to pass the degree here: We could add a function to fixed_data or to Analyzed where you pass a `&[PolyID]` (or an iterator or whateven) and it returns the common degree of the columns or an error.

Reply:
Changed to this thanks!!